eng/pipelines/templates/jobs/archetype-sdk-client.yml

```diff
@@ -146,7 +146,7 @@ jobs:
       - task: ms.vss-governance-buildtask.governance-build-task-component-detection.ComponentGovernanceComponentDetection@0
         # ComponentGovernance is currently unable to run on pull requests of public projects. Running on non-PR
         # builds should be sufficient.
-        condition: and(succeededOrFailed(), ne(variables['Build.Reason'], 'PullRequest'))
+        condition: and(succeeded(), ne(variables['Build.Reason'], 'PullRequest'))
```

Comment:
Why skip ComponentGovernance if an earlier task failed (say Verify Readmes)?  Usually with analyze/test tasks, it's better to run as much as possible even if an earlier step failed.  I agree it should not be `always()` since we want to skip on cancel, but I think `succeededOrFailed()` is better than just `succeeded()`.

Reply:
My thinking here was that if we've got a failed job, the validity of the component scan could be invalid anyway, so why bother doing it. But happy to do ```succeededOrFailed()``` in this case since it won't run if we cancel.